### ISSUE TITLE
InvalidParameterValueException not thrown in glacier.uploadArchive when checksum is invalid

### ISSUE THREAD

**Issue by trivikr** [aws/aws-sdk-js-v3#851]
**Describe the bug**
InvalidParameterValueException not thrown in glacier.uploadArchive when checksum is invalid

**SDK version number**
```console
$ yarn list aws-sdk
└─ aws-sdk@2.609.0
$ yarn list @aws-sdk/client-glacier
└─ @aws-sdk/client-glacier@1.0.0-alpha.11
```

**Is the issue in the browser/Node.js?**
Node.js

**Details of the browser/Node.js version**
v12.14.0

**To Reproduce (observed behavior)**
* Tested in branch `glacier-checksum-test` https://github.com/trivikr/aws-sdk-js-tests/tree/glacier-checksum-test
* File: https://github.com/trivikr/aws-sdk-js-tests/blob/285b94bf24f6475bb7dd5a78aafc5c7be9bb38f1/src/node.js

Code:
```javascript
const AWS = require("aws-sdk");
const { Glacier } = require("@aws-sdk/client-glacier");
const { REGION } = require("./config");

(async () => {
  const v2Client = new AWS.Glacier({ region: REGION });
  const v3Client = new Glacier({ region: REGION });

  const vaultName = "TestVault";

  const body = Buffer.alloc(0.25 * 1024 * 1024);
  body.fill("0");
  const incorrectChecksum = "00000000000000000000000000000000";

  await v2Client.createVault({ vaultName }).promise();
  await v2Client.waitFor("vaultExists", { vaultName }).promise();

  try {
    await v2Client
      .uploadArchive({
        vaultName,
        body,
        checksum: incorrectChecksum
      })
      .promise();
  } catch (error) {
    console.log(`Error from v2:`);
    console.log(JSON.stringify(error, null, 2));
  }

  try {
    await v3Client.uploadArchive({
      vaultName,
      body,
      checksum: incorrectChecksum
    });
  } catch (error) {
    console.log(`\nError from v3:`);
    console.log(JSON.stringify(error, null, 2));
  }
})();
```

Output:
```console
Error from v2:
{
  "message": "Checksum mismatch: expected 0000000000000000000000000000000000000000000000000000000000000000, computed 6faefade5a638cd3545d638dd5754763658e32209e69420cb559b7650d4bf93a",
  "code": "InvalidParameterValueException",
  "time": "2020-02-05T00:20:19.490Z",
  "requestId": "t9E_uo6BnAxMzMqrgeVLsMfzCxPjQGLPUv7-AH0QOxcbUNM",
  "statusCode": 400,
  "retryable": false,
  "retryDelay": 91.80764884047183
}

Error from v3:
{}
```

**Expected behavior**
v3 should throw InvalidParameterValueException error

**Comment by trivikr** [comment]
Rechecking with `@aws-sdk/client-glacier@1.0.0-alpha.15`

The error thrown in v3 is `UnknownError` and not `InvalidParameterValueException`

Output:
```
Error from v2:
InvalidParameterValueException: Checksum mismatch: expected 0000000000000000000000000000000000000000000000000000000000000000, computed 6faefade5a638cd3545d638dd5754763658e32209e69420cb559b7650d4bf93a
    at Object.extractError (/Users/trivikr/workspace/aws-sdk-js-tests/node_modules/aws-sdk/lib/protocol/json.js:51:27)
    at Request.extractError (/Users/trivikr/workspace/aws-sdk-js-tests/node_modules/aws-sdk/lib/protocol/rest_json.js:55:8)
    at Request.callListeners (/Users/trivikr/workspace/aws-sdk-js-tests/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/Users/trivikr/workspace/aws-sdk-js-tests/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/Users/trivikr/workspace/aws-sdk-js-tests/node_modules/aws-sdk/lib/request.js:683:14)
    at Request.transition (/Users/trivikr/workspace/aws-sdk-js-tests/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/Users/trivikr/workspace/aws-sdk-js-tests/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /Users/trivikr/workspace/aws-sdk-js-tests/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/Users/trivikr/workspace/aws-sdk-js-tests/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/Users/trivikr/workspace/aws-sdk-js-tests/node_modules/aws-sdk/lib/request.js:685:12) {
  message: 'Checksum mismatch: expected 0000000000000000000000000000000000000000000000000000000000000000, computed 6faefade5a638cd3545d638dd5754763658e32209e69420cb559b7650d4bf93a',
  code: 'InvalidParameterValueException',
  time: 2020-02-07T22:42:42.664Z,
  requestId: '_oemzL8-VpqdIpvS3gcCUrLafKjTuCZAfjxEC2kloO85WWo',
  statusCode: 400,
  retryable: false,
  retryDelay: 35.17965710191222
}

Error from v3:
Error [UnknownError]: Checksum mismatch: expected 0000000000000000000000000000000000000000000000000000000000000000, computed 6faefade5a638cd3545d638dd5754763658e32209e69420cb559b7650d4bf93a
    at deserializeAws_restJson1_1UploadArchiveCommandError (/Users/trivikr/workspace/aws-sdk-js-tests/node_modules/@aws-sdk/client-glacier/dist/cjs/protocols/Aws_restJson1_1.js:2938:41)
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {
  code: 'InvalidParameterValueException',
  type: 'Client',
  name: 'UnknownError',
  '$fault': 'client',
  '$metadata': {
    httpStatusCode: 400,
    httpHeaders: {
      'x-amzn-requestid': 'g-t5VSsnPk1HvLBanzZVL5yhwY2__AbN8gZGOnckmRIuZRk',
      'content-type': 'application/json',
      'content-length': '237',
      date: 'Fri, 07 Feb 2020 22:42:42 GMT',
      connection: 'close'
    },
    requestId: 'g-t5VSsnPk1HvLBanzZVL5yhwY2__AbN8gZGOnckmRIuZRk',
    retries: 0,
    totalRetryDelay: 0
  }
}
```

* Tested in branch `glacier-checksum-test` https://github.com/trivikr/aws-sdk-js-tests/tree/glacier-checksum-test
* File: https://github.com/trivikr/aws-sdk-js-tests/blob/d1aa8f1cacb44fa571ac9bb9906893ecdb90f5ef/src/node.js

**Comment by trivikr** [comment]
This happens because `output.headers` don't have `"x-amzn-errortype"`
https://github.com/aws/aws-sdk-js-v3/blob/a10e47019abd2e255e72f4462ec67f88b236eab7/clients/client-glacier/protocols/Aws_restJson1_1.ts#L4224-L4227

Because of this, name is set to `"UnknownError"`
https://github.com/aws/aws-sdk-js-v3/blob/a10e47019abd2e255e72f4462ec67f88b236eab7/clients/client-glacier/protocols/Aws_restJson1_1.ts#L4266-L4273

**Comment by srchase** [comment]
Confirmed fix in [@aws-sdk/client-glacier@1.0.0-alpha.19](https://www.npmjs.com/package/@aws-sdk/client-glacier/v/1.0.0-alpha.19)

Code:
```
const { Glacier } = require("@aws-sdk/client-glacier");

(async () => {
    const v3Client = new Glacier({ region: "us-west-2" });
  
    const vaultName = "TestV3";
  
    const body = Buffer.alloc(0.25 * 1024 * 1024);
    body.fill("0");
    const incorrectChecksum = "00000000000000000000000000000000";
  
    try {
      await v3Client.uploadArchive({
        vaultName,
        body,
        checksum: incorrectChecksum
      });
    } catch (error) {
      console.log(error);
    }
  })();
```

Response:
```
InvalidParameterValueException: Checksum mismatch: expected 0000000000000000000000000000000000000000000000000000000000000000, computed 6faefade5a638cd3545d638dd5754763658e32209e69420cb559b7650d4bf93a
    at deserializeAws_restJson1_1UploadArchiveCommandError...
  code: 'InvalidParameterValueException',
  type: 'Client',
  name: 'InvalidParameterValueException',
  '$fault': 'client',
  '$metadata': {
    httpStatusCode: 400,
    httpHeaders: {
      'x-amzn-requestid': '...',
      'content-type': 'application/json',
      'content-length': '237',
      date: 'Fri, 14 Feb 2020 18:37:31 GMT',
      connection: 'close'
    },
    requestId: '...',
    retries: 0,
    totalRetryDelay: 0
  }
}
```
